backend/src/main/java/capstone/myfinancemanager/manager/controller/TransactionController.java

```diff
@@ -26,9 +31,15 @@ public List<TransactionDto> getAllTransactions() {
     @ResponseStatus(code = HttpStatus.CREATED)
     @PostMapping
     public ResponseEntity<TransactionDto> addTransaction(@RequestBody TransactionCreationDto transactionCreation) {
+
+        Principal principal = SecurityContextHolder.getContext().getAuthentication();
```

Comment:
Spring will inject the principal if you include it in the method signature like so `addTransaction(Principal principal, @RequestBody TransactionCreationDto transactionCreation)`